beets/library.py

```diff
@@ -760,6 +760,8 @@ def write(self, path=None, tags=None, id3v23=None):
         mediafile.update(item_tags)
         try:
             mediafile.save()
+            # Move file to trigger OS event if some processes subscribed to it
+            util.move(path, path, force=True)
```

Comment:
Why is this necessary/should we do this? I guess this is forcing an update on file timestamps (is that actually guaranteed to happen, or is it just an implementation detail of the file system that this was tested on?). My first thought here would be that we should leave timestamp updated to the file system/mount option; and I would expect an mtime update to happen here, anyway.

Reply:
Let me explain the case when `move` is required.

For example you're using wifi connection to work with your library which is stored as huge `flac` files, and you would like to insert a new and cool cover arts.

Music file can be big. Very big. dozen of megabytes. Cover arts with nice resolution also might be a few megs.

So, writing it via wifi isn't something fast. It requires time. Some time.

From another hand you may use smart NAS which indexes your music collection when it changed.

I do have such setup with Synology as NAS, and it starts to index a file when it appears. If file appears not fast enough => it corrupt it's index.

This move triggered forced re-index of file and `move` on the same FS is atomic, and it fires all listener who would like to know when file is updated *after* it is moved to distination.

Reply:
Ok, that's the issue that you reported elsewhere. I don't like the solution to be honest, it seems like a crude workaround that most people won't need, and it doesn't even prevent the intermediate broken index. A possible solution that I just thought of would be to use the [hook plugin](https://beets.readthedocs.io/en/stable/plugins/hook.html) to `touch` files in the `after_write` hook.

Reply:
@wisp3rwind touch isn't enough. File should be moved away ;)

Reply:
Alternatively, it might be possible to implement proper atomic behaviour in `mediafile`.

Reply:
@wisp3rwind but it will be the same hack, isn't it?

Reply:
> @wisp3rwind touch isn't enough. File should be moved away ;)

Well, you could do that in the hook, too. And there's still the issue that I don't think there's any guarantee that `os.replace(path, path)` does anything at all. The python 3.9 docs don't state this. In fact, if this is using `rename(3)` under the hood, POSIX even specifies
>If the old argument and the new  ar‐
       gument resolve to either the same existing directory entry or different directory entries for the same
       existing file, rename() shall return successfully and perform no other action.

Reply:
> @wisp3rwind but it will be the same hack, isn't it?

No, the idea would be that `mediafile` created a new file (with a temporary suffix `.beets`) which would be renamed when it is fully written.

Reply:
@wisp3rwind I agree with your point to move it inside `mediafile`

Reply:
or inside `mutagen`

Reply:
and it has the issue: https://github.com/quodlibet/mutagen/issues/241 :)